packages/world/ts/config/v2/world.ts

```diff
@@ -75,22 +76,40 @@ export function resolveWorld<const world extends WorldInput>(world: world): reso
   const scope = extendedScope(world);
   const namespaces = world.namespaces ?? {};
 
+  const resolvedStore = resolveStore({
+    ...world,
+    codegen: mergeIfUndefined(world.codegen ?? {}, {
+      // put codegen into namespaced directories if we're using `namespaces` key
+      namespaceDirectories: world.namespaces != null,
+      // don't generate `index.sol` if we're using `namespaces` key, to avoid naming conflicts for table names across namespaces
+      // this is a deprecated option anyway, so this default guides folks towards not using it
+      indexFilename: world.namespaces != null ? false : undefined,
```

Comment:
this comment mentions that this is deprecated but I don't think we actually deprecated this yet, can we?

Reply:
How do we mark something as deprecated?

Link to original discussion https://github.com/latticexyz/mud/pull/2880#discussion_r1622475589

Reply:
```ts
// @deprecated
```
or
```ts
/**
 * @deprecated
 */
```